Index: week3/oop_cars/cars.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- week3/oop_cars/cars.py	(revision f64504560b156ffe408b57c96bd59c741fc137d1)
+++ week3/oop_cars/cars.py	(date 1543525587000)
@@ -1,8 +1,13 @@
 import os.path
 import csv
 
+SPEC_MACHINE = 'spec_machine'
+TRUCK = 'truck'
+CAR = 'car'
+
+
 class CarBase:
-    def __init__(self, brand, photo_file_name, carrying, car_type = None):
+    def __init__(self, brand, photo_file_name, carrying):
         self.brand = brand
         self.photo_file_name = photo_file_name
         self.carrying = float(carrying)
@@ -12,11 +17,15 @@
         extention = os.path.splitext(self.photo_file_name)[1]
         return extention
 
+    def __repr__(self):
+        return "{" + self.brand + " " + self.photo_file_name + " " + str(self.carrying) + " " + self.car_type + "}"
+
+
 class Car(CarBase):
     def __init__(self, brand, photo_file_name, carrying, passenger_seats_count):
         super().__init__(brand, photo_file_name, carrying)
         self.passenger_seats_count = int(passenger_seats_count)
-        self.car_type = 'car'
+        self.car_type = CAR
 
 
 class Truck(CarBase):
@@ -25,40 +34,60 @@
         if body_whl:
             self.body_width, self.body_height, self.body_length = [float(x) for x in body_whl.split('x')]
         else:
-            self.body_width, self.body_height, self.body_length = [0,0,0]
-        self.car_type = 'truck'
+            self.body_width, self.body_height, self.body_length = [0, 0, 0]
+        self.car_type = TRUCK
 
     def get_body_volume(self):
         return self.body_width*self.body_height*self.body_length
 
+
 class SpecMachine(CarBase):
     def __init__(self, brand, photo_file_name, carrying, extra):
         super().__init__(brand, photo_file_name, carrying)
         self.extra = extra
-        self.car_type = 'spec_machine'
+        self.car_type = SPEC_MACHINE
 
 
 def get_car_list(csv_filename):
     car_list = []
-    with open(csv_filename, encoding="utf8") as f:
-        reader = csv.reader(f, delimiter=';')
+    with open(csv_filename, encoding="utf8") as file:
+        reader = csv.reader(file, delimiter=';')
         next(reader)
         for row in reader:
-            try:
-                if row[0] == 'car':
-                    car_list.append(Car(row[1], row[3], row[5], row[2]))
-                elif row[0] == 'truck':
-                    car_list.append(Truck(row[1], row[3], row[5], row[4]))
-                elif row[0] == 'spec_machine':
-                    car_list.append(SpecMachine(row[1], row[3], row[5], row[6]))
-                else:
-                    pass
-            except IndexError:
-                pass
+            if len(row) < 7:
+                print('invalid row: ', row)
+                break
+
+            car = build_car(row)
+            if car:
+                car_list.append(car)
+
     return car_list
 
+
+def build_car(row):
+    car_type = row[0]
+    brand = row[1]
+    passenger_seats_count = row[2]
+    photo_file_name = row[3]
+    body_whl = row[4]
+    carrying = row[5]
+    extra = row[6]
+    if car_type == CAR:
+        car = Car(brand, photo_file_name, carrying, passenger_seats_count)
+    elif car_type == TRUCK:
+        car = Truck(brand, photo_file_name, carrying, body_whl)
+    elif car_type == SPEC_MACHINE:
+        car = SpecMachine(brand, photo_file_name, carrying, extra)
+    else:
+        print('unknown car type', car_type)
+        car = None
+    return car
+
+
 def _main():
-    get_car_list("coursera_week3_cars.csv")
+    print('Result: ', get_car_list("coursera_week3_cars.csv"))
+
 
 if __name__ == "__main__":
     _main()
Index: week3/oop_cars/coursera_week3_cars.csv
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- week3/oop_cars/coursera_week3_cars.csv	(revision f64504560b156ffe408b57c96bd59c741fc137d1)
+++ week3/oop_cars/coursera_week3_cars.csv	(date 1543525098000)
@@ -3,5 +3,6 @@
 truck;Man;;f2.png;8x3x2.5;20;
 truck;Man;;f2.png;;20;
 car;Mazda 6;4;f3.jpeg;;2.5;
+bad_car;Mazda 6;4;f3.jpeg;;2.5;
 ;;;
 spec_machine;Hitachi;;f4;;1.2;Легкая техника для уборки снега
